### DISCUSSION TITLE
MathJax/KaTeX support

### DISCUSSION THREAD

**Discussion by tholzschuh** [srid/emanote#213] [Ideas]
Is it possible to use MathJax/KaTeX within emanote?

I tried adding `<script async="" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>` like in neuron to the headHtml, but it doesn't seem to work..

What am I doing wrong?

**Comment by srid** [comment]
I'm not familiar with MathJax JS but they should work. Did you check your browser JS console for any errors? Try using a verbose form of the MathJax include and/or include the JS snippet near the end of body by overriding `after-note.tpl`; cf. https://github.com/srid/emanote/discussions/131

**Reply by srid** [reply]
@tholzschuh working demo https://uiuc-cs-zettelkasten.github.io

source https://github.com/uiuc-cs-zettelkasten/uiuc-cs-zettelkasten.github.io

**Reply by srid** [reply]
They seem to include the `<script>` tag inline, but that's not ideal. If you can make it work, we can add it to Emanote docs, under here https://note.ema.srid.ca/tips/js

**Reply by tholzschuh** [reply]
@srid so I played around a little bit and what confuses me is the following:
If I refresh the page, I get to see the correct "math rendering" for a split second before it doesn't work anymore for some reason.
If I rename my `index.yaml` file for a second and then rename it back, the MathJax gets correctly displayed.
Only upon refreshing the page, it gets "broken" again.

I'll try using a verbose version later today!

**Reply by srid** [reply]
Probably ema's web socket page replace is getting in the way. Did you check the browser console for errors?

You shouldn't see these issues in statically generated site (`emanote gen`), I'm sure.

**Reply by tholzschuh** [reply]
Yeah, `emanote gen` works!
Or more precisely: The MathJax now works, but all the links are broken.

Why does emanote gen include ` <base href='/' />` everywhere?
Shouldn't it set the base href to the folder I'm giving it as the output location?

// Edit: nevermind, I just found the baseUrl option in index.yaml :-)

**Reply by srid** [reply]
It's up to you to specify a baseUrl in your index.yaml, otherwise Emanote will use the default from here:

https://github.com/srid/emanote/blob/243f8dc5424cd2050b694cc4805cb3e1f595b812/default/index.yaml#L30-L31

(In general you can override any of the attrs in this file)

**Comment by srid** [comment]
This has been added Emanote:

- https://github.com/srid/emanote/pull/489 (thanks to @reptee)
- https://emanote.srid.ca/tips/js/math#katex
